> That's unusual. When I run the same snippet of code you posted, I get the error: Py3.10, v0.0.232
> 
> ```
> 
> Traceback (most recent call last):
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/urllib3/connection.py", line 174, in _new_conn
>     conn = connection.create_connection(
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/urllib3/util/connection.py", line 72, in create_connection
>     for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
>   File "/Users/me/opt/anaconda3/envs/py310/lib/python3.10/socket.py", line 955, in getaddrinfo
>     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
> socket.gaierror: [Errno 8] nodename nor servname provided, or not known
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/urllib3/connectionpool.py", line 714, in urlopen
>     httplib_response = self._make_request(
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/urllib3/connectionpool.py", line 403, in _make_request
>     self._validate_conn(conn)
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/urllib3/connectionpool.py", line 1053, in _validate_conn
>     conn.connect()
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/urllib3/connection.py", line 363, in connect
>     self.sock = conn = self._new_conn()
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/urllib3/connection.py", line 186, in _new_conn
>     raise NewConnectionError(
> urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x1326e8c10>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/requests/adapters.py", line 486, in send
>     resp = conn.urlopen(
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/urllib3/connectionpool.py", line 798, in urlopen
>     retries = retries.increment(
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/urllib3/util/retry.py", line 592, in increment
>     raise MaxRetryError(_pool, url, error or ResponseError(cause))
> urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='docs.pinecone.iohttps', port=443): Max retries exceeded with url: //www.pinecone.io/ (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1326e8c10>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/Users/me/Library/Application Support/JetBrains/PyCharm2023.1/scratches/loader_snippet.py", line 5, in <module>
>     docs = loader.load()
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/langchain/document_loaders/recursive_url_loader.py", line 102, in load
>     return list(self.lazy_load())
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/langchain/document_loaders/recursive_url_loader.py", line 96, in lazy_load
>     child_links = self.get_child_links_recursive(self.url)
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/langchain/document_loaders/recursive_url_loader.py", line 88, in get_child_links_recursive
>     visited.update(self.get_child_links_recursive(link, visited))
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/langchain/document_loaders/recursive_url_loader.py", line 62, in get_child_links_recursive
>     response = requests.get(url)
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/requests/api.py", line 73, in get
>     return request("get", url, params=params, **kwargs)
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/requests/api.py", line 59, in request
>     return session.request(method=method, url=url, **kwargs)
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
>     resp = self.send(prep, **send_kwargs)
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
>     r = adapter.send(request, **kwargs)
>   File "/Users/me/code/snippets/.vector-db/lib/python3.10/site-packages/requests/adapters.py", line 519, in send
>     raise ConnectionError(e, request=request)
> requests.exceptions.ConnectionError: HTTPSConnectionPool(host='docs.pinecone.iohttps', port=443): Max retries exceeded with url: //www.pinecone.io/ (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1326e8c10>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))
> ```
> 
> Sorry about the linter failure, will fix in the AM.

Oh, interesting! Hmmm ... `Python 3.9.16` and current `master` for me.

I fixed lint. 

Still, this is a good change:

`urljoin(base_url, link)` is a more robust way to construct URLs if link is a relative path, an absolute path, or a fully qualified URL and it handles cases where the base URL doesn't end with a slash, or the link starts with a slash.

--- a/tests/unit_tests/document_loaders/test_recursive_url_loader.py
+++ b/tests/unit_tests/document_loaders/test_recursive_url_loader.py
@@ -40,6 +40,10 @@ def mock_requests_get(monkeypatch: MonkeyPatch) -> None:
     mock_response_absolute = MagicMock()
     mock_response_absolute.text = "Absolute page"
 
+    # Mock Response object for default
+    mock_response_default = MagicMock()
+    mock_response_default.text = "Default page"
+
     def mock_get(url: str, *args: Any, **kwargs: Any) -> Mock:
         if url.startswith("http://test.com"):
             if "/absolute" in url:
@@ -48,6 +52,7 @@ def mock_requests_get(monkeypatch: MonkeyPatch) -> None:
                 return mock_response_relative
             else:
                 return mock_response_main
+        return mock_response_default
 
     monkeypatch.setattr(
         "langchain.document_loaders.recursive_url_loader.requests.get", mock_get

